File: ejercicios-js/ejercicio5.js
/**
 * Genera un numero aleatorio entre 1 y 100
 *
 * @return un numero aleatorio entre 0 y 100
 */
function randomNumber() {
    return Math.floor(Math.random() * 100) + 1;
}

/**
 * Determina si un numero es primo
 * @param {number} number el numero a evaluar
 * @returns {boolean} true si el numero es primo, false si no lo es
 */
function isPrime(number) {
    if (number <= 1) {
        return false;
    }

    for (let i = 2; i * i <= number; i++) {
        if (number % i === 0) {
            return false;
        }
    }


    return true;
}


const generatedNumber = randomNumber();

console.log(`El número ${generatedNumber} ${isPrime(generatedNumber) ? "" : "no "}es primo`);

File: ejercicios-py/ejercicio5.py
import random

"""
Genera un numero aleatorio entre 1 y 100

:return: un numero aleatorio entre 0 y 100
"""
def random_number():
    return random.randint(1, 100)

"""
Determina si un numero es primo
:param number: el numero a evaluar
:return: True si el numero es primo, False si no lo es
"""
def is_prime(number):
    if number <= 1:
        return False

    index = 2
    while index * index <= number:
        if number % index == 0:
            return False

        index += 1

    return True


generated_number = random_number()

print(f"El número {generated_number} {'' if is_prime(generated_number) else 'no '}es primo")